strongback/src/org/strongback/command/CommandGroup.java

```diff
@@ -151,15 +151,19 @@ private CommandGroup(Command[] commands, Type type) {
     }
 
     Type getType() {
-        return type;
+        if (root != null) {
+            return root.type;
+        } else {
+            return type;
+        }
     }
 
     Command[] getCommands() {
-        return commands;
-    }
-
-    Command getRoot() {
-        return root;
+        if (root != null) {
+            return root.getCommands();
+        } else {
+            return commands;
+        }
```

Comment:
Same minor nit ... since the if block returns, there is no need for an 'else' block, so it'd be better to do this:

    if (root != null) {
        return root.getCommands();
    }
    return commands;

or, better yet:

    return root != null ? root.getCommands() : command;